### DISCUSSION TITLE
Where is elasticsearch configuration file in securityonion2?

### DISCUSSION THREAD

**Discussion by dahalsulav** [Security-Onion-Solutions/securityonion#9474] [General]
I have installed security onion2 in my ubuntu 20. I want to configure elasticsearch snapshot, however I am unable to find the exact elasticsearch.yml location in order to add 'repo' path.
Note: I have already tried configuring elasticsearch.yml (inside /opt/so/conf/elasticsearch) but the changes automatically reset after the elasticsearch gets restarted. Also, I have tried configure defaults.yml inside saltstack, but it failed to restart the elasticsearch soon I make changes. 
What could be the possible solution for it?

**Comment by alexandruhera** [comment]
All of the components run in Docker containers, the above path wouldn't work. I am having the same issue, but I haven't decided to mess with the container files or see if there is a 'proper' way to do it.

**Comment by TotieBash** [comment]
"but the changes automatically reset after the elasticsearch gets restarted"
If you "vim /opt/so/saltstack/default/salt/elasticsearch/init.sls" you will see that it is guarded by Salt.  The file it is referencing is located in /opt/so/saltstack/default/salt/elasticsearch/files/elasticsearch.yaml.jinja
![image](https://user-images.githubusercontent.com/86748904/209859666-5d52521d-2ecb-4bbb-8f2e-828363444c7b.png)

What you do is copy the file from "default" to "local" and edit the local file.
cp /opt/so/saltstack/default/salt/elasticsearch/files/elasticsearch.yaml.jinja /opt/so/saltstack/local/salt/elasticsearch/files/elasticsearch.yaml.jinja

The file is in jinja format and that is where the end of my help.  My understanding of jinja file format is zero.
![image](https://user-images.githubusercontent.com/86748904/209860339-c7216997-7ef2-4e35-9491-342275c486d0.png)

**Comment by TotieBash** [comment]
I am not comfortable with this answer below but if I was in your situation and I am desperate to try something then the below is what I would try on my lab environment.  This "should" make your edit to  /opt/so/conf/elasticsearch/elasticsearch.yml to be honored and survive reboot.

cp /opt/so/saltstack/default/salt/elasticsearch/init.sls /opt/so/saltstack/local/salt/elasticsearch/init.sls

vim /opt/so/saltstack/local/salt/elasticsearch/init.sls and comment out lines189-197
![image](https://user-images.githubusercontent.com/86748904/209861211-7c4e1492-b615-4e74-8085-e74aaf9eef49.png)

Good luck

**Comment by TotieBash** [comment]
As far as the Elasticsearch snapshot, have you tried it using the official instruction? If not, then I think this is the supported method to do it hence this should be your first option.
https://docs.securityonion.net/en/2.3/backups.html?highlight=elasticsearch#elasticsearch
![image](https://user-images.githubusercontent.com/86748904/209865162-3edcc673-dc35-4415-bbd2-f25bd0ed7c57.png)
